adaptors/extensions.js

```diff
@@ -21,3 +21,35 @@ exports.clearTemporaryFile = function clearTemporaryFile(req, reply) {
 
   reply.continue();
 };
+
+exports.logRequest = function logRequest(req, reply) {
+  // We don't want to clutter the terminal, so only
+  // show request details if this was an error
+  if (req.response.isBoom) {
+    var errorData = req.response.data;
+    var errorObj = errorData && errorData.error;
+
+    // Prefer the error object stack over the
+    // boom object stack
+    var stack = errorObj && errorObj.stack;
+    stack = stack ? stack : req.response.stack;
+
+    if (!req.response.data || req.response.data.debug) {
+      req.log.debug({
+        request: req,
+        response: req.response,
+        error: req.response,
+        stack: stack
+      });
+    } else {
+      req.log.error({
+        request: req,
+        response: req.response,
+        error: req.response,
```

Comment:
No, it just moves that logic here since we don't want to log the error if `data` doesn't exist. I think it's cleaner to leave it as-is